sqlgen/matcher_test.go

```diff
@@ -0,0 +1,50 @@
+package sqlgen
+
+import (
+	"sort"
+	"testing"
+
+	"github.com/kylelemons/godebug/pretty"
+)
+
+func TestMatcher(t *testing.T) {
+	m := newMatcher()
+
+	m.add(Filter{"foo": "bar", "id": 1}, 0)
+	m.add(Filter{"id": 1}, 1)
+	m.add(Filter{"foo": "baz", "id": 1}, 2)
+	m.add(Filter{}, 3)
```

Comment:
I changed the test to be more clear. I renamed the opaque value to id so that matcher becomes much more like a map. I think it's more understandable now, and your comment was helpful. Thanks.